applications/IgaApplication/tests/cpp_tests/test_iga_shell_3p.cpp

```diff
@@ -0,0 +1,325 @@
+        const std::vector<double> expected_LHS_row_1({133490.278045005,850779.615701094,0,26281.0006695998,121138.251571485,0,1940.28454630975,5520.78640580566,0,63.6658342060302,68.8450946538420,0,0.783391240126120,-0.534780027131337,0,-133490.278045005,-794755.404416132,0,-26281.0006695998,-168890.105746305,0,-1940.28454630975,-13385.9868564238,0,-63.6658342060302,-469.321814260949,0,-0.783391240126120,-6.14515988952213,0});
+        const std::vector<double> expected_LHS_row_2({0,0,11846.6051205351,0,0,-10105.0283553663,0,0,-1656.23498738675,0,0,-83.9478154806674,0,0,-1.39396230134425,0,0,-11837.3538952984,0,0,10087.4365759178,0,0,1663.68728170558,0,0,84.8136642417877,0,0,1.41637343323110});
+        const std::vector<double> expected_right_hand_side_vector({0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0});
+        
+        KRATOS_CHECK_VECTOR_NEAR(row(left_hand_side_matrix,0), expected_LHS_row_0, tolerance);
+        KRATOS_CHECK_VECTOR_NEAR(row(left_hand_side_matrix,1), expected_LHS_row_1, tolerance);
+        KRATOS_CHECK_VECTOR_NEAR(row(left_hand_side_matrix,2), expected_LHS_row_2, tolerance);
+        KRATOS_CHECK_VECTOR_NEAR(right_hand_side_vector, expected_right_hand_side_vector, tolerance);
+    }
+
+    // Tests the stiffness matrix of the Shell3pElement with a polynomial degree of p=5.
+    KRATOS_TEST_CASE_IN_SUITE(IgaShell3pElementP5, KratosIgaFastSuite)
+    {
+        Model current_model;
+        auto &r_model_part = current_model.CreateModelPart("ModelPart");
+        r_model_part.GetProcessInfo().SetValue(DOMAIN_SIZE, 3);
+
+        r_model_part.AddNodalSolutionStepVariable(DISPLACEMENT);
+
+        IntegrationPoint<3> integration_point(0.0337652428984240, 0.211324865405187, 0, 0.0428311230947926);
+        auto p_shell_3p_element = GetShell3pElement(r_model_part, 5, integration_point);
+
+        TestCreationUtility::AddDisplacementDofs(r_model_part);
+
+        p_shell_3p_element->Initialize();
+
+        Matrix left_hand_side_matrix;
+        Vector right_hand_side_vector;
+        p_shell_3p_element->CalculateLocalSystem(left_hand_side_matrix, right_hand_side_vector, r_model_part.GetProcessInfo());
+
+        //Check RHS and LHS results
+        const double tolerance = 1.0e-1;
+
+        //Matrix expected_left_hand_side_matrix = ZeroMatrix(36,36);
+        const std::vector<double> expected_LHS_row_0({405000.290031407,123985.676772767,0,-33973.9881469315,-106654.871970357,0,-9694.63076821954,-16422.3627864133,0,-594.585433000862,-887.278239041373,0,-14.8585189361368,-20.9788839821913,0,-0.135084710956110,-0.184892973551875,0,-276681.912058038,33221.8619642891,0,-76407.9570133629,-28578.0868133018,0,-7301.61436342488,-4400.35884643036,0,-323.698899579919,-237.745487612845,0,-6.85347582290271,-5.62127502113436,0,-0.0562693799549848,-0.0495419229494137,0});
```

Comment:
since you know the size better use `std::array` (always use `std::array` if the size is known, its faster, I recommend google for the reasons :) )

Reply:
Thanks we changed it